From 1b4dcee3119f32449e010225cfb8b8b5fce13281 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 2 Apr 2018 19:33:34 +0200
Subject: [PATCH 1/1] ARM: vexpress: fdtfile, fdt_addr_r for vexpress_ca15_tc2

Provide environment values for fdtfile and fdt_addr_r for the
Versatile Express V2P-CA15-CA7 (TC2).

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 configs/vexpress_ca15_tc2_defconfig | 1 +
 include/configs/vexpress_common.h   | 9 ++++++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/configs/vexpress_ca15_tc2_defconfig b/configs/vexpress_ca15_tc2_defconfig
index f04aa7c104..7009b253c5 100644
--- a/configs/vexpress_ca15_tc2_defconfig
+++ b/configs/vexpress_ca15_tc2_defconfig
@@ -4,6 +4,7 @@ CONFIG_SYS_TEXT_BASE=0x80800000
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_BOOTCOMMAND="run distro_bootcmd; run bootflash"
+CONFIG_DEFAULT_FDT_FILE="vexpress-v2p-ca15_a7.dtb"
 # CONFIG_DISPLAY_CPUINFO is not set
 # CONFIG_DISPLAY_BOARDINFO is not set
 # CONFIG_CMD_CONSOLE is not set
diff --git a/include/configs/vexpress_common.h b/include/configs/vexpress_common.h
index 3e2d410f14..b3ca9af70a 100644
--- a/include/configs/vexpress_common.h
+++ b/include/configs/vexpress_common.h
@@ -178,7 +178,8 @@
 #ifdef CONFIG_VEXPRESS_ORIGINAL_MEMORY_MAP
 #define CONFIG_PLATFORM_ENV_SETTINGS \
 		"loadaddr=0x80008000\0" \
-		"ramdisk_addr_r=0x61000000\0" \
+		"fdt_addr_r=0x61000000\0" \
+		"ramdisk_addr_r=0x61800000\0" \
 		"kernel_addr=0x44100000\0" \
 		"ramdisk_addr=0x44800000\0" \
 		"maxramdisk=0x1800000\0" \
@@ -188,7 +189,8 @@
 #elif defined(CONFIG_VEXPRESS_EXTENDED_MEMORY_MAP)
 #define CONFIG_PLATFORM_ENV_SETTINGS \
 		"loadaddr=0xa0008000\0" \
-		"ramdisk_addr_r=0x81000000\0" \
+		"fdt_addr_r=0x81000000\0" \
+		"ramdisk_addr_r=0x81800000\0" \
 		"kernel_addr=0x0c100000\0" \
 		"ramdisk_addr=0x0c800000\0" \
 		"maxramdisk=0x1800000\0" \
@@ -209,7 +211,8 @@
 			"devtmpfs.mount=0  vmalloc=256M\0" \
 		"bootflash=run flashargs; " \
 			"cp ${ramdisk_addr} ${ramdisk_addr_r} ${maxramdisk}; " \
-			"bootm ${kernel_addr} ${ramdisk_addr_r}\0"
+			"bootm ${kernel_addr} ${ramdisk_addr_r}\0" \
+		"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0"
 
 /* FLASH and environment organization */
 #define PHYS_FLASH_SIZE			0x04000000	/* 64MB */
-- 
2.16.2

